logger.js

```diff
@@ -50,28 +52,39 @@ function Logger(opts) {
 
     this.statsd = opts.statsd;
 
-    var levels = extend(defaultLevels, opts.levels || {});
-    var path = opts.path || "";
+    var levels = this.levels = extend(defaultLevels, opts.levels || {});
+    this.path = opts.path = "";
 
-    Object.keys(levels)
-        .forEach(function (levelName) {
+    this.streamsByLevel = Object.keys(levels)
+        .reduce(function (streamsByLevel, levelName) {
             if (!levels[levelName]) {
-                return;
+                return streamsByLevel;
             }
 
             var level = levels[levelName];
             level = extend({transforms: []}, level);
+            levels[levelName] = level;
```

Comment:
On line 70 @kriskowal is using {} as the this scope which I think means that mutating levels here doesn't mutate the this.levels. However I think the code could be more transparent.